proofs/alf/cvc5/Cvc5.smt3

```diff
@@ -189,33 +189,68 @@
   :conclusion (= t (run_evaluate t))
 )
 
-(program $is_poly_norm ((T Type) (U Type) (a T) (b T) (a1 U) (a2 U) (b1 U) (b2 U))
-  (T T) Bool
-  (
-    (($is_poly_norm (> a1 a2) (> b1 b2))   ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (>= a1 a2) (>= b1 b2)) ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (<= a1 a2) (<= b1 b2)) ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (< a1 a2) (< b1 b2))   ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    (($is_poly_norm (= a1 a2) (= b1 b2))   (alf.ite (is_arith_type (alf.typeof a1)) 
-                                           (alf.ite (is_arith_type (alf.typeof b1))
-                                             ($is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) false)
-                                             false)
-                                           false)) ; TODO: bv
-    (($is_poly_norm a b)                   (alf.ite (is_arith_type (alf.typeof a)) 
-                                           (alf.ite (is_arith_type (alf.typeof b))
-                                             ($is_poly_norm_arith a b)
-                                             false)
-                                           false)) ; TODO: bv
-  )
-)
-
-; ProofRule::ARITH_POLY_NORM
+; rule: arith_poly_norm
+; implements: ProofRule::ARITH_POLY_NORM
+; args:
+; - a U: The left-hand side of the equality.
+; - b U: The right-hand side of the equality.
+; requires: >
+;   That the arithmetic/bitvector expressions a and b are equal up polynomial
+;   normalization.
+; note: >
+;   The current implementation of $is_poly_norm does not support bitvectors.
+;   Therefore, cvc5 will not output this rule for bitvector proofs; instead, it
+;   will output a trust rule.
+; conclusion: The given equality.
 (declare-rule arith_poly_norm ((U Type) (a U) (b U))
   :args ((= a b))
   :requires ((($is_poly_norm a b) true))
   :conclusion (= a b)
 )
 
+; define: $is_poly_norm_rel
+; args:
+; - r (-> U U Bool): The binary relation to check.
+; - cx U: The scaling factor for (- x1 x2).
```

Comment:
What is x1,x2,y1,y2 here?  I think this relates to below but it should be stated here.